# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSink.h
# Commit: c632760187c5
# Full Hash: c632760187c5c0d69adcff405050d8f233ebca1d
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-24 21:44:48
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Trigger the audibility event at the beginning of the initialization of an audio stream, when un-muting. r=alwu
#   
#   The initialization itself can take some time and this audibility event is used
#   to prioritize the process, that will soon become audible, on at least Windows,
#   so it's better to get prioritized slightly ahead of time.
# ==============================================================================

diff -r 7f259ef8550f -r c632760187c5 dom/media/mediasink/AudioSink.h
--- a/dom/media/mediasink/AudioSink.h	Tue May 24 13:09:09 2022 +0000
+++ b/dom/media/mediasink/AudioSink.h	Tue May 24 13:09:10 2022 +0000
@@ -26,6 +26,11 @@
 
 class AudioSink : private AudioStream::DataSource {
  public:
+  enum class InitializationType {
+    // This AudioSink is being initialized for the first time
+    INITIAL,
+    UNMUTING
+  };
   struct PlaybackParams {
     PlaybackParams(double aVolume, double aPlaybackRate, bool aPreservesPitch)
         : mVolume(aVolume),
@@ -42,7 +47,8 @@
   ~AudioSink();
 
   // Allocate and initialize mAudioStream. Returns NS_OK on success.
-  nsresult InitializeAudioStream(const PlaybackParams& aParams);
+  nsresult InitializeAudioStream(const PlaybackParams& aParams,
+                                 InitializationType aInitializationType);
 
   // Start audio playback.
   nsresult Start(const media::TimeUnit& aStartTime,